modules/renter/renter.go

```diff
@@ -338,6 +339,29 @@ func (r *Renter) ProcessConsensusChange(cc modules.ConsensusChange) {
 	r.mu.Unlock(id)
 }
 
+// validateSiapath checks that a Siapath is a legal filename.
+// ../ is disallowed to prevent directory traversal, and paths must not begin
+// with / or be empty.
+func validateSiapath(siapath string) error {
+	if strings.HasPrefix(siapath, "/") {
+		return errors.New("siapath cannot begin with /")
+	}
+
+	if siapath == "" {
+		return ErrEmptyFilename
+	}
+
+	if strings.Contains(siapath, "../") {
```

Comment:
isn't this a superset of the next .Contains()? And, I think the rule is a bit more complicated (explained this in another PR) - what's not allowed is to have a full file that's named '..' or '.', but you can have a './' in the name if for example your folder is named 'foo.', then you might have a file 'foo./bar', which should be legal.

Instead we want to check for the prefix '../' and the prefix './', as well as the contains '/../' and '/./'.